src/components/engagement/engagement.service.ts

```diff
@@ -1560,4 +1575,13 @@ export class EngagementService {
       ).every((n) => n)
     );
   }
+
+  protected async getProjectTypeById(projectId: string): Promise<string> {
+    const qr = `
+    MATCH (p:Project {id: $projectId, active: true}) RETURN p.type as type
+    `;
+    const results = await this.db.query().raw(qr, { projectId }).first();
+
+    return results?.type;
```

Comment:
```suggestion
    return results?.type as ProjectType | undefined;
```
This is better and more correct than `string`

Reply:
Yes, I've changed.